src/sinks/util/auto_concurrency/tests.rs

```diff
@@ -630,7 +630,7 @@ async fn all_tests() {
     // The first delay takes just slightly longer than all the rest,
     // which causes the first test to run differently than all the
     // others. Throw in a dummy delay to take up this delay "slack".
-    let _ = tokio::spawn(async move { delay_for(Duration::from_millis(1)) }).await;
```

Comment:
@bruceg clippy caught the fact that we were not awaiting on the `delay_for` here. I'm wondering if that had something to do with the weird timing issue you saw?

I believe I fixed this correctly by simply returning the future from `delay_for` rather than including an `async {}` block; I felt this better than just adding an `.await` in the block.

Reply:
Rather, hmm, I see this line was added specifically for that issue. In this case I don't think it would have waited at all though so I'm curious what side-effects it has.

Reply:
I'm getting a bit of a Twilight Zone feeling. I have no idea why both these lines are in there. I thought the delay required a separate task since the clock was paused from this task, but removing the `tokio::spawn` line completely produced the same result… or at least it does now.

Reply:
I know that feeling. Do you want me to remove the spawn and the comment above it?

Reply:
Correct, just drop the spawn, leave the real delay.